JS/index.js

```diff
@@ -200,23 +216,27 @@ function isGameOver() {
 async function createWaves() {
     for (let i = 0; GAME_STATE === GAME_STATE_ACTIVE && i < 10; i++) {
         for (let times = 0; GAME_STATE === GAME_STATE_ACTIVE && times < 4; times++) {

```

Comment:
You validate game state in "for" cycle setup and and then also check "isGameOver" in "if" within a cycle.
It makes not sense, leaving just "if" is enough.

Reply:
fin